replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/MetacardConfigLoader.java

```diff
@@ -54,97 +54,80 @@
+  }
+
+  @VisibleForTesting
+  MetacardConfigLoader(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      Security security) {
     this.framework = framework;
     this.filterBuilder = filterBuilder;
     this.helper = helper;
-    this.configMetacardType = configMetacardType;
     this.siteManager = siteManager;
+    this.newConfigManager = newConfigManager;
+    this.security = security;
   }
 
-  public ReplicatorConfig createConfig() {
-    return new ReplicatorConfigImpl();
+  public void init() {
+    Failsafe.with(
+            new RetryPolicy()
+                .withBackoff(3, 60, TimeUnit.SECONDS)
+                .withMaxDuration(10, TimeUnit.MINUTES))
+        .onFailedAttempt(
+            throwable -> LOGGER.debug("failed to upgrade legacy configs, will retry", throwable))
+        .onFailure(
+            throwable ->
+                LOGGER.debug("failed to upgrade legacy configs, restart to try again.", throwable))
+        .run(this::convertConfigs);
   }
 
-  public ReplicatorConfig getConfig(final String id) {
-    List<ReplicatorConfig> configs =
-        helper.getTypeForFilter(
-            filterBuilder.allOf(
-                filterBuilder.attribute(ReplicationConfig.NAME).is().equalTo().text(id),
-                filterBuilder
-                    .attribute(Metacard.TAGS)
-                    .is()
-                    .like()
-                    .text(ReplicationConfig.METACARD_TAG)),
-            this::getConfigFromMetacard);
-
-    final int count = configs.size();
-    if (count == 1) {
-      return configs.get(0);
-    } else if (count > 1) {
-      throw new IllegalStateException(
-          "Found more than one replication config for id {}. This should never happen");
-    } else {
-      throw new NotFoundException("Unable to find replication config for id " + id);
-    }
+  private void convertConfigs() {
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  getAllConfigs()
+                      .forEach(
+                          config -> {
+                            newConfigManager.save(config);
+                            removeConfig(config);
+                          });
+                  return null;
+                });
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.warn(
+                "Error occured while trying to transform catalog saved metacards to persistent store saved configs. Legacy configs may not be accessible.",
+                e);
+          }
+          return null;
+        });
   }
 
-  public List<ReplicatorConfig> getAllConfigs() {
+  List<ReplicatorConfig> getAllConfigs() {
     return helper.getTypeForFilter(
```

Comment:
:question: Is there a way here to differentiate errors from unavailability to reach the data?